src/mono/wasm/runtime/assets.ts

```diff
@@ -427,6 +427,85 @@ function _instantiate_asset(asset: AssetEntry, url: string, bytes: Uint8Array) {
     endMeasure(mark, MeasuredBlock.instantiateAsset, asset.name);
     ++actual_instantiated_assets_count;
 }
+type ManifestRow = [name: string, length: number]
+function mount_archive_VFS(prefix: string, manifest: ManifestRow[], data: Uint8Array) {
+    const anyModule = Module as any;
+    const FS = anyModule.FS;
+    const read_only_mode = 0o0444; //-r--r--r--
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    const noop = () => { };
+    const dummyTime = () => 0;
+    const ARCHIVEFS = {
+        mount: function () {
+            const root = ARCHIVEFS.createNode(null, "/", read_only_mode);
+            root.children = {};
+            return root;
+        },
+        createNode: (parent: any, name: string, mode: number) => {
+            const node = FS.createNode(parent, name, mode);
+            node.node_ops = ARCHIVEFS.node_ops;
+            node.stream_ops = ARCHIVEFS.stream_ops;
+            if (parent && parent.children) parent.children[name] = node;
+            return node;
+        },
+        node_ops: {
+            getattr: (node: any) => {
+                return {
+                    mode: read_only_mode,
+                    dev: 0,
+                    ino: 0,
+                    nlink: 0,
+                    uid: 0,
+                    gid: 0,
+                    rdev: 0,
+                    size: node.length,
+                    atime: { getTime: dummyTime },
+                    mtime: { getTime: dummyTime },
+                    ctime: { getTime: dummyTime },
+                    blksize: 0,
+                    blocks: 0
+                };
+            },
+            lookup: (parent: any, name: string) => {
+                if (!parent || !parent.children) return null;
+                return parent.children[name];
+            },
+        },
+        stream_ops: {
+            read: function (stream: any, buffer: Int8Array, offset: number, length: number, position: number) {
+                const toRead = Math.min(stream.node.length - position, length);
+                const source = data.subarray(stream.node.offset + position, stream.node.offset + position + toRead);
+                Module.HEAPU8.set(source, offset);
+                return toRead;
+            },
+            open: noop,
+            close: noop,
+            llseek: noop
+        }
+    };
+    FS.mkdirTree(prefix);
+    const root = FS.mount(ARCHIVEFS, { root: "." }, prefix);
+
+    let offset = 0;
+    for (const row of manifest) {
+        const name = row[0];
+        const length = row[1];
```

Comment:
Row[1] == length of the path? How do we know it? Is the manifest a file that I could peek into?

Reply:
The manifest never hits the file system in JSON form. See `src\tasks\WasmBuildTasks\GenerateWasmBundle.cs` how it's created. If you open `artifacts\bin\microsoft.netcore.app.runtime.browser-wasm\Debug\runtimes\browser-wasm\native\dotnet.timezones.blat` it's pretty much readable.